src/main/java/BankingSystem.java

```diff
@@ -0,0 +1,214 @@
+import dao.AccountDao;
+import model.Account;
+import model.Card;
+
+import java.util.Scanner;
+
+class BankingSystem {
+
+    private final AccountDao accountDao;
+    private final CardGenerator cardGenerator;
+    private final Scanner scanner;
+    private Account currentAccount;
+
+    public BankingSystem(AccountDao accountDao) {
+        this.accountDao = accountDao;
+        cardGenerator = new CardGenerator();
+        scanner = new Scanner(System.in);
+    }
+
+    public void showMenu() {
+        while (true) {
+            welcomeMenu();
+
+            String input = scanner.next();
+
+            switch (input) {
+                case "0" -> stopApplication();
+                case "1" -> registerAccount();
+                case "2" -> logIntoAccount();
+                default -> System.out.println("You've entered invalid menu item.\n");
+            }
+        }
+    }
+
+    private void welcomeMenu() {
+        System.out.println("""
+                1. Create an account
+                2. Log into account
+                0. Exit""");
+    }
+
+    private void registerAccount() {
+        Card card = cardGenerator.generateCredentials();
+        accountDao.save(new Account(card, 0));
+        System.out.printf("""
+                Your card has been created
+                Your card number:
+                %s
+                Your card PIN:
+                %s""", card.number(), card.pin());
+
+        System.out.print("\n".repeat(2));
+    }
+
+    private void logIntoAccount() {
+        System.out.println("\nEnter your card number:");
+        String inCardNum = scanner.next();
+        System.out.println("Enter your PIN:");
+        String inCardPIN = scanner.next();
+
+        accountDao.get(inCardNum, inCardPIN).ifPresentOrElse(account -> {
+                    currentAccount = account;
+                    System.out.println("\nYou have successfully logged in!\n");
+                    accountMenu();
+                },
```

Comment:
That's looking better now, good job! But it can look even better)
Can you please move it to a separate method
```
account -> {
       currentAccount = account;
       System.out.println("\nYou have successfully logged in!\n");
       accountMenu();
}
```
then call it using method reference. It should look something like this
```
accountDao.get(inCardNum, inCardPIN).ifPresentOrElse(
             this::doSomething,
             () -> System.out.println("\nWrong card number or PIN!\n"));
```

Reply:
Done